### PR TITLE
Add script to import code docs from curriculumbuilder

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Been working on this on and off for a while and I think code docs is finally in a good place that this can be reviewed. Based on our other import scripts, this one runs through all the programming expressions on levelbuilder, grabs the data on curriculumbuilder, saves the expressions with that data, then serializes the expression.

This will be run on levelbuilder in a couple of weeks (not sure exactly when). 

## Testing story

tested locally

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/curriculum/import_code_docs.rb [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+      exp.name = cb_exp['title']
+      exp.content = cb_exp['content']
+      exp.short_description = cb_exp['description']
+      exp.external_documentation = cb_exp['ext_doc']
+      exp.return_value = cb_exp['return_value']
+      exp.tips = cb_exp['tips']
+      if cb_exp['image']
+        log "#{exp.key} has an image at #{cb_exp['image']}"
+        exp.image_url = cb_exp['image']
```
Comment (tim-dot-org) [inline bin/curriculum/import_code_docs.rb:93]: are these already hosted somewhere like s3, or is there additional work to move images and video?

File: bin/curriculum/import_code_docs.rb [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      end
+      exp.examples = cb_exp['examples']
+      exp.palette_params = cb_exp['parameters']
+      exp.save!
+
+      category = exp.programming_environment.categories.find_by_name(cb_exp['category'])
+      exp.programming_environment_category = category if category
+      warn "#{exp.key} is in a non-existant category #{cb_exp['category']}" unless category
+
+      exp.serialize
```
Comment (tim-dot-org) [inline bin/curriculum/import_code_docs.rb]: is serialize what you want, or `write_serialization`?

File: bin/curriculum/import_code_docs.rb [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      if cb_exp['image']
+        log "#{exp.key} has an image at #{cb_exp['image']}"
+        exp.image_url = cb_exp['image']
+      end
+      if cb_exp['video']
+        log "#{exp.key} has a video with URL #{cb_exp['video']}"
+      end
+      exp.examples = cb_exp['examples']
+      exp.palette_params = cb_exp['parameters']
+      exp.save!
```
Comment (tim-dot-org) [inline bin/curriculum/import_code_docs.rb]: should this save be moved after the category assignment below?

File: bin/curriculum/import_code_docs.rb (hunk 1 above)
Comment (bethanyaconnor) [reply bin/curriculum/import_code_docs.rb:93]: These should be saved on s3 and will stick around when curriculumbuilder is turned off. We could move them as part of the contractor pass, but not sure if we will this time around

File: bin/curriculum/import_code_docs.rb (hunk 3 above)
Comment (bethanyaconnor) [reply bin/curriculum/import_code_docs.rb]: I don't think it matters but I can move this down

File: bin/curriculum/import_code_docs.rb (hunk 2 above)
Comment (bethanyaconnor) [reply bin/curriculum/import_code_docs.rb]: good catch! that would explain why I didn't have a bunch of .json changes hanging around like I expected....

File: bin/curriculum/import_code_docs.rb [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+      exp.serialize
+    end
+  end
+end
+
+def fetch(url)
+  uri = URI(url)
+  response = Net::HTTP.get_response(uri)
+  #raise "HTTP status #{response.code} fetching #{uri}" unless response.is_a? Net::HTTPSuccess
```
Comment (davidsbailey) [inline bin/curriculum/import_code_docs.rb]: remove debug code?

**Review by davidsbailey — APPROVED** [review]
Looks good to me!

**Review by tim-dot-org — APPROVED** [review]

**Pull request merged**